Src/RE2.Managed/Regex2.cs

```diff
@@ -273,14 +273,16 @@ public static unsafe bool Matches(string pattern, string text, out List<Dictiona
                                 submatchString = Encoding.UTF8.GetString(textUtf8Bytes, submatchTextStartIndex, submatchLength);
                             }
 
+                            var flexMatch = new FlexMatch() { Success = true, Index = submatchTextStartIndex, Length = submatchLength, Value = submatchString };
```

Comment:
>submatchTextStartIndex [](http://example.com/codeflow?start=86&length=22)

This is the start index in a byte buffer which represents a UTF-8 encoded string.

It is not necessarily the same as the start index of the Unicode character in a Unicode string.